platforms/software/build-init/src/main/java/org/gradle/buildinit/plugins/internal/VersionCatalogDependencyRegistry.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.buildinit.plugins.internal;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.util.internal.TextUtil;
+
+import java.util.Collection;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@NonNullApi
+public class VersionCatalogDependencyRegistry {
```

Comment:
🤔  This could be package private, me thinks

Reply:
used from Maven conversion code